tests/test_apply.py

```diff
@@ -1022,3 +1022,30 @@ def test_pauliz_hadamard(self, theta, phi, varphi, monkeypatch, qubit_device_3_w
             - 2 * np.cos(theta) * np.sin(phi) * np.sin(2 * varphi)
         ) / 4
         assert np.allclose(var, expected, atol=tol, rtol=0)
+
+
+class TestTooManyQubits:
+    """Test for when too many wires are requested, where lightning.qubit should behave like
+    default.qubit."""
+
+    def test_warning(self):
+        """Test that a warning is produced when lightning.qubit is instantiated with more than
+        the supported number of qubits"""
+        with pytest.warns(UserWarning, match="The number of wires exceeds"):
+            qml.device("lightning.qubit", wires=LightningQubit._MAX_WIRES + 1)
+
+    def test_apply(self, monkeypatch):
+        """Test that the apply method uses the one in default.qubit when the number of wires
+        exceeds the supported number."""
+        dev = qml.device("lightning.qubit", wires=2)
+        dev.num_wires = LightningQubit._MAX_WIRES + 1
+        mock_apply = mock.MagicMock()
+        with monkeypatch.context() as m:
+            m.setattr(DefaultQubit, "apply", mock_apply)
+            dev.apply(0, rotations=1, other=2)
+        call_args = mock_apply.call_args
```

Comment:
We've tried avoiding the use of `unittest` and `mock` where possible and keep tests `pytest` only. One way to do that would be to create a `MockApply` class:
```python
class MockApply:
    def __init__(self):
        self.op_call = None
        self.rotations_call = None
        self.kwargs_call = None

    def apply(self, operations, rotations=None, **kwargs):
        self.op_call = operations
        self.rotations_call = rotations
        self.kwargs_call = kwargs
```
and then
```python
mock_apply = MockApply()
with monkeypatch.context() as m:
        m.setattr(DefaultQubit, "apply", mock_apply.apply)
```
(or comparable, this one is not tested code)

Reply:
Nice, thanks @antalszava I will update that.

Out of curiosity, what motivates the drive to avoid `unittest`?

Reply:
I think we'd just want to stick with a single testing framework whenever possible and have one less dependency if possible :thinking: Perhaps @josh146 would have further thoughts here

Reply:
@antalszava @trbromley: Note that `unittest.mock` has no pytest equivalent, so it's usually fine to use `unittest.mock` with pytest (better to use an existing package than roll our own mocking framework).

Having said that, why not just use [`pytest-mock`](https://pypi.org/project/pytest-mock/)? This is a simple wrapper around `unittest.mock` to turn it into a fixture (and make it fit better with common pytest patterns):

```python
def test_apply(self, mocker):
    """Test that the apply method uses the one in default.qubit when the number of wires
    exceeds the supported number."""
    mocker.patch("pennylane.devices.DefaultQubit.apply")

    dev = qml.device("lightning.qubit", wires=2)
    dev.num_wires = LightningQubit._MAX_WIRES + 1
    dev.apply(0, rotations=1, other=2)

    DefaultQubit.apply.assert_called_once_with(0, rotations=1, other=2)
```

`pytest.mock` also suports `mocker.spy`, and adheres to the `unittest.mock` api.

Reply:
Thanks a lot to both of you for this! I've updated to @josh146's suggestion - which is also clearer!